Examples/VisionExamples/Examples/SafetyAlertsViewController.swift

```diff
@@ -0,0 +1,184 @@
+        let documentsPath =
+            NSSearchPathForDirectoriesInDomains(.documentDirectory,
+                                                .userDomainMask,
+                                                true).first!
+        let path = documentsPath.appending("/safety-alerts-drawing")
+
+        // create VisionReplayManager with a path to recorded session
+        visionManager = try? VisionReplayManager.create(recordPath: path)
+        // register its delegate
+        visionManager.delegate = self
+
+        // create VisionSafetyManager and register as its delegate to receive safety related events
+        visionSafetyManager = VisionSafetyManager.create(visionManager: visionManager)
+        // register its delegate
+        visionSafetyManager.delegate = self
+
+        // configure Vision view to display sample buffers from video source
+        visionViewController.set(visionManager: visionManager)
+        // add Vision view as a child view
+        addVisionView()
+
+        // add view to draw overspeeding alert
+        addOverspeedingAlertView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        visionManager.start()
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+
+        visionManager.stop()
+    }
+
+    deinit {
+        // free up VisionManager's resources
+        visionManager.destroy()
+    }
+
+    private func addVisionView() {
+        addChild(visionViewController)
+        view.addSubview(visionViewController.view)
+        visionViewController.didMove(toParent: self)
+    }
+
+    private func addOverspeedingAlertView() {
+        alertOverspeedingView = UIImageView(image: UIImage(named: "alert"))
+        alertOverspeedingView.isHidden = true
+        alertOverspeedingView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(alertOverspeedingView)
+        NSLayoutConstraint.activate([
+            alertOverspeedingView.topAnchor.constraint(equalToSystemSpacingBelow: view.topAnchor, multiplier: 1),
+            view.trailingAnchor.constraint(equalToSystemSpacingAfter: alertOverspeedingView.trailingAnchor, multiplier: 1)
+        ])
+    }
+
+    // MARK: - Drawing custom collision detection views
+
+    private func clearCollisionStateDrawings() {
+        for subview in view.subviews {
+            if subview.isKind(of: CollisionDetectionView.self) {
+                subview.removeFromSuperview()
+            }
+        }
+    }
+
+    // MARK: - Handle VisionSafety events
+
+    private func updateCollisionDrawing() {
+        clearCollisionStateDrawings()
+
+        for carCollision in carCollisions {
+            // Calculate absolute coordinates
+            let relativeBBox = carCollision.lastDetection.boundingBox
+            let frameSize = carCollision.lastFrame.image.size.cgSize
+
+            let bbox = CGRect(x: relativeBBox.origin.x * frameSize.width,
```

Comment:
At this stage, `bbox` has the coordinates in the camera frame space. Users should convert it to the view space saving aspect ratio.